apps/mobile/src/libs/components/common/loader-wrapper/loader-wrapper.tsx

```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import { ActivityIndicator } from "react-native";
+
+import { View } from "~/libs/components/components";
+import { BaseColor } from "~/libs/enums/app/base-color.enum";
+import { globalStyles } from "~/libs/styles/styles";
+
+type Properties = {
+	children: React.ReactNode;
+	isLoading: boolean;
+};
+
+const LoaderWrapper: React.FC<Properties> = ({ children, isLoading }) => {
+	const renderContent = () => {
+		if (isLoading) {
+			return <ActivityIndicator color={BaseColor.BG_BLUE} size="large" />;
+		}
+		return children;
+	};
+
+	return (
+		<View style={[globalStyles.flex1, globalStyles.justifyContentCenter]}>
+			{renderContent()}
+		</View>
+	);
+};
```

Comment:
You may not want to affect the layout of the child component. And I believe we can get rid of the additional wrapper for ActivityIndicator.
```suggestion
	return isLoading ? (
		  <ActivityIndicator 
		      color={BaseColor.BG_BLUE}
		      size="large"
		      style={[
		          globalStyles.flex1,
		          globalStyles.justifyContentCenter,
		          globalStyles.alignItemsCenter
		      ]} />
	) : (
		<>{children}</>
	);
};
```